### PR TITLE
Solved #6640 Replaced ```markdox``` with ```jsdoc-to-markdown```

### PR DESCRIPTION
**Pull request by abhishek-bodapati**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)
Fixed issue #6640. Replaced markdox with jsdoc-to-markdown

### CODE REVIEW

File: scripts/docs-generation/wdioDocs.js [hunk 1]
```diff
@@ -1,18 +1,9 @@
 const fs = require('fs')
 const path = require('path')
-const markdox = require('markdox')
-const { promisify } = require('util')
-
-const formatter = require('../utils/formatter')
-const compiler = require('../utils/compiler')
```
Comment (christian-bromann) [inline scripts/docs-generation/wdioDocs.js:7]: Are these files still needed?

File: scripts/docs-generation/wdioDocs.js [hunk 2]
```diff
@@ -1,18 +1,9 @@
...
+const jsdoc2md = require('jsdoc-to-markdown')
 const TEMPLATE_PATH = path.join(__dirname, '..', 'templates', 'api.tpl.ejs')
```
Comment (christian-bromann) [inline scripts/docs-generation/wdioDocs.js:4]: What happened to this template file? How do we ensure that the docs are still in the right format?

Comment (abhishek-bodapati) [reply scripts/docs-generation/wdioDocs.js:4]: Made changes..!!
Please have a look

File: scripts/docs-generation/wdioDocs.js [hunk 3]
```diff
@@ -35,12 +43,11 @@ exports.generateWdioDocs = async (sidebars) => {
 
             const filepath = path.join(COMMAND_DIR, scope, file)
             const outputpath = path.join(docDir, `_${file.replace(/(js|ts)/, 'md')}`)
-            const out = await processDocs({data:[filepath],template:TEMPLATE_PATH })
+            const out = await processDocs({data:[filepath],template:TEMPLATE_PATH, options:JSDOC_OPTIONS})
```
Comment (abhishek-bodapati) [inline scripts/docs-generation/wdioDocs.js:46]: Added JSDOC_OPTIONS

File: scripts/docs-generation/wdioDocs.js (hunk 2 above)
Comment (abhishek-bodapati) [reply scripts/docs-generation/wdioDocs.js:4]: Added 
```
const JSDOC_OPTIONS = {
    formatter: formatter,
    compiler: compiler,
    template: TEMPLATE_PATH
}
```
and included the options 
```const out = await processDocs({data:[filepath], options:JSDOC_OPTIONS}) ```

These must ensure the generated ```.md``` files are in the right format..!!

Let me know if any changes are to be done!

Comment (christian-bromann) [reply scripts/docs-generation/wdioDocs.js:4]: @abhishek-bodapati can you build the page and send a screenshot?

File: scripts/docs-generation/wdioDocs.js (hunk 3 above)
Comment (christian-bromann) [inline scripts/docs-generation/wdioDocs.js]: Are you sure both packages have the same interface?

**Review by christian-bromann — COMMENTED** [review]
Please note that you can't just replace the packages in the code. `markdox` has its own interface and the replacement for it needs to support similar features.

File: scripts/docs-generation/wdioDocs.js (hunk 2 above)
Comment (abhishek-bodapati) [reply scripts/docs-generation/wdioDocs.js:4]: I had run ```node generateDocs.js ``` to generate ```.md``` files from ```.ts``` files. Build was successful. Here are the screenshots 
Screenshot 2021-05-25 at 7 44 18 PM
Screenshot 2021-05-25 at 7 44 28 PM

File: scripts/docs-generation/wdioDocs.js (hunk 3 above)
Comment (abhishek-bodapati) [reply scripts/docs-generation/wdioDocs.js]: Yeah! The build is successfully compiled and the ```.md``` files are generated without any errors and warnings

**Review by christian-bromann — COMMENTED** [review]
Couple of things:

- `markdox` is still in the `package.json`, `jsdoc-to-markdown` would need to be added
- all generated files are empty which causes website build to fail due to

        Error: Bad sidebars file.
        These sidebar document ids do not exist:

File: scripts/docs-generation/wdioDocs.js (hunk 3 above)
Comment (christian-bromann) [reply scripts/docs-generation/wdioDocs.js]: it is failing for me, see comment below

**Comment by abhishek-bodapati** [comment]
> Couple of things:
> 
> * `markdox` is still in the `package.json`, `jsdoc-to-markdown` would need to be added
> * all generated files are empty which causes website build to fail due to
>   ```
>     Error: Bad sidebars file.
>     These sidebar document ids do not exist:
>   ```

Yeah! Understood the problem, will work on it!
Thanks for letting me know

**Comment by christian-bromann** [comment]
@abhishek-bodapati any updates?

**Comment by abhishek-bodapati** [comment]
> @abhishek-bodapati any updates?

No

**Comment by christian-bromann** [comment]
@abhishek-bodapati are you planning to continue working on this?

**Comment by abhishek-bodapati** [comment]
> @abhishek-bodapati are you planning to continue working on this?

No @christian-bromann

**Comment by christian-bromann** [comment]
Ok, closing then.
